---
title: "Visualizing Smaller Variants"
author: "Scott Furlan"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 3
params:
  EVAL: !r identical(Sys.getenv("NOT_CRAN"), "true")
vignette: >
  %\VignetteIndexEntry{Visualizing Smaller Variants}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
alias: annotating_variants
editor_options: 
  chunk_output_type: console
---

```{r, child="children/SETTINGS-knitr_long.txt"}
```

## Plotting SnpEff Variants

Plotting the number of variants by annotation category.  One needs to first annotate variants using SnpEff.  Generating this plot may take a while depending of the number of variants you have.

```{r, eval=FALSE}
library(pacbiowdlR)
variantsfile <- "location/small_variants.ann.txt.gz"
plot_snpEff(variantsfile, cores = 16)
```

![](../man/figures/variants.png){width=100%}
```{r, eval=FALSE, include = FALSE}
data_wsu <- read_json_file("/Volumes/furlan_s/sfurlan/pbjay/WSU_AML_ctl/pacbiowdlR/outputs.json")
variantsfile <- data_wsu$humanwgs_singleton.tertiary_small_variant_filtered_vcf

vcf <- VariantAnnotation::readVcf(variantsfile)
mutsfile  <- system.file("extdata", "examples/wsuaml_mutations.csv", package = "pacbiowdlR")
#muts <- read.csv("/Users/sfurlan/develop/pacbiowdlR/inst/extdata/examples/wsuaml_mutations.csv")
muts <- read.csv(mutsfile)
mutsGR <- GRanges(seqnames = muts$Chromosome, IRanges(start = muts$Position, end = muts$Position))
#but this is hg19 probs.
# Import chain, run liftOver
library(rtracklayer)
library(magrittr)
utils::download.file(
  url = "http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz",
  destfile = "hg19ToHg38.over.chain.gz",
  mode = "wb"
)
system(paste("gunzip -c", shQuote("hg19ToHg38.over.chain.gz"), ">", shQuote("hg19ToHg38.over.chain")))
chain <- import.chain("hg19ToHg38.over.chain")
lifted <- liftOver(mutsGR, chain)
muts_grch38 <- unlist(lifted)
  
ol <- findOverlaps(vcf@rowRanges, muts_grch38)
fvcf <- vcf[ol@from,]
#okay not hg19


ol <- findOverlaps(vcf@rowRanges, mutsGR)
fvcf <- vcf[ol@from,]
VariantAnnotation::alt(fvcf)

#now filter for common
mutsGR_filt <- mutsGR[ol@to,]
muts_filt <- muts[ol@to,]

tab <- table(muts_filt$Alt.Allele==as.character(VariantAnnotation::alt(fvcf) %>% unlist()))

dim(fvcf)[1]/nrow(muts)
tab[2]/(tab[1]+tab[2]) %>% as.numeric()
```


```{r, eval=FALSE, include = FALSE}



S5 <- read_json_file("/Volumes/furlan_s/sfurlan/250520_leuklong/pbWGS/LL2_S5/pacbiowdlR/outputs.json")
variantsfile <- S5$humanwgs_singleton.phased_small_variant_vcf
variantsfile <- S5$humanwgs_singleton.tertiary_small_variant_filtered_vcf

vcf <- VariantAnnotation::readVcf(variantsfile)
muts <- read.csv("/Volumes/furlan_s/sfurlan/250520_leuklong/snpEff/snpEff_out/LL2_S5/LL2_S5.small_variants.ann.txt", sep = "\t")

#muts <- read.csv("/Volumes/furlan_s/sfurlan/250520_leuklong/snpEff/snpEff_out/LL2_S5/LL2_S5.small_variants.clinvar.txt", sep = "\t")

muts[muts$CLNSIG=="Pathogenic" & !is.na(muts$CLNSIG),]

mi <- muts[grep("IKZF1", muts$ANN),]

mi[mi$TYPE=="INDEL",]
```
