---
title: "Analyzing, Annotating, and Plotting DNA Fusions"
author: "Scott Furlan"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 3
params:
  EVAL: !r identical(Sys.getenv("NOT_CRAN"), "true")
vignette: >
  %\VignetteIndexEntry{Analyzing, Annotating, and Plotting DNA Fusions}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
alias: dna_fusions
editor_options: 
  chunk_output_type: console
---

```{r, child="children/SETTINGS-knitr_fusions.txt"}
```




```{r, eval=TRUE}
library(pacbiowdlR)
sv_file  <- system.file("extdata", "examples/K562.GRCh38.structural_variants.phased.vcf.gz", package = "pacbiowdlR")

fusions<-get_fusions_from_vcf(sv_file, plot = T, filter = "")
```


Wow, that is a lot of fusions.  Many of these may be artifact, but K562s are genomically complex to say the least.  Let's see if we can learn a little more about these events.  First, to properly annotate them we need to load a gtf file into memory.  This annotation uses the "canonical transcripts" for every gene, which is simply defined for these purposes as the longest transcript.  Note that for human genome, this can take up about 10GB in RAM and takes about 1-2 minutes. One can check the presence of a loaded gtf as shown.

```{r, eval=TRUE}
gtf <- "/Users/sfurlan/Library/CloudStorage/OneDrive-FredHutchCancerCenter/computation/refs/GTFs/gencode.v47.chr_patch_hapl_scaff.annotation.gtf"
preload_gtf(gtf)
exists("gtf_env", .GlobalEnv)
```

Then we can annotate the fusions  (dont worry the gtf is loaded and won't need to reload if the above steps worked):

```{r, eval=TRUE}
fusions <- get_fusions_from_vcf(sv_file, gtf_file = gtf, plot = T, annotate = T, filter = "")

```

Okay, that is a seemingly overwhelming number of gene-gene fusions...  We can filter the fusion calls to known fusions and highlight the fusions that are found in the database.

```{r, eval=TRUE}
db_fusions <- get_neosplice_db()

results <- find_related_fusions(fusions, db_fusions)
hits <- as.numeric(which(sapply(results, function(f) length(f)>0)))
fusions[hits,]
print_fusion_matches(results[hits])

plot_fusions(fusions, annotate = T, highlight = hits)
```



```{r}

a4573 <- read_json_file("/Volumes/furlan_s/grp/data/FurlanLab2025/lewings/pbWGS/A4573/pacbiowdlR/outputs.json", sub1 = "/fh/fast")
tc71 <- read_json_file("/Volumes/furlan_s/grp/data/FurlanLab2025/lewings/pbWGS/TC71/pacbiowdlR/outputs.json", sub1 = "/fh/fast")
tc32 <- read_json_file("/Volumes/furlan_s/grp/data/FurlanLab2025/lewings/pbWGS/TC32/pacbiowdlR/outputs.json", sub1 = "/fh/fast")
chla10 <- read_json_file("/Volumes/furlan_s/grp/data/FurlanLab2025/lewings/pbWGS/CHLA10/pacbiowdlR/outputs.json", sub1 = "/fh/fast")
chla9 <- read_json_file("/Volumes/furlan_s/grp/data/FurlanLab2025/lewings/pbWGS/CHLA9/pacbiowdlR/outputs.json", sub1 = "/fh/fast")


fusions<-get_fusions_from_vcf(a4573$humanwgs_singleton.phased_sv_vcf, plot = T, filter = "", thresh = 10, highlight = 28, title = "A4573")
fusions<-get_fusions_from_vcf(tc71$humanwgs_singleton.phased_sv_vcf, plot = T, filter = "", thresh = 5, highlight = 69, title = "TC71")
fusions<-get_fusions_from_vcf(tc32$humanwgs_singleton.phased_sv_vcf, plot = T, filter = "", thresh = 5, highlight = 25, title = "TC32")
fusions<-get_fusions_from_vcf(chla9$humanwgs_singleton.phased_sv_vcf, plot = T, filter = "", thresh = 5, highlight = 16, title = "CHLA9")
fusions<-get_fusions_from_vcf(chla10$humanwgs_singleton.phased_sv_vcf, plot = T, filter = "", thresh = 1, title = "CHLA10")
```



```{r, eval=F, include = F}
data_wsu <- read_json_file("/Volumes/furlan_s/sfurlan/pbjay/WSU_AML_ctl/pacbiowdlR/outputs.json")

gtf <- "/Users/sfurlan/Library/CloudStorage/OneDrive-FredHutchCancerCenter/computation/refs/GTFs/gencode.v47.chr_patch_hapl_scaff.annotation.gtf"
preload_gtf(gtf)
fusions <- get_fusions_from_vcf(data_wsu$humanwgs_singleton.phased_sv_vcf, gtf_file = gtf, plot = T, annotate = T, filter = "", thresh = 3)
fusions$fusion_details

fusions[6,]
```
