<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>CNA Plot with Gene Highlighting — CNAPlot_Highlight • pacbiowdlR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous"><link href="../docsearch.css" rel="stylesheet"><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="CNA Plot with Gene Highlighting — CNAPlot_Highlight"><meta property="og:description" content="Generates a genome-wide CNA plot with additional gene annotation and labeling.
This function extends CNAPlot by annotating gene coordinates (using
annotateCoverageWithGenes()) and optionally highlighting selected genes
and cytogenetic (G-band) features."><meta property="og:image" content="https://scfurl.github.io/pacbiowdlR/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pacbiowdlR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>

  </a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/scfurl/pacbiowdlR" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
<li>
  <a href="https://bsky.app/profile/scfurl.bsky.social" class="external-link">
    <span class="fa fa-bluesky"></span>

  </a>
</li>
      </ul><form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off"></div>
      </form>

    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>CNA Plot with Gene Highlighting</h1>

    <div class="hidden name"><code>CNAPlot_Highlight.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generates a genome-wide CNA plot with additional gene annotation and labeling.
This function extends <code>CNAPlot</code> by annotating gene coordinates (using
<code><a href="annotateCoverageWithGenes.html">annotateCoverageWithGenes()</a></code>) and optionally highlighting selected genes
and cytogenetic (G-band) features.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">CNAPlot_Highlight</span><span class="op">(</span></span>
<span>  <span class="va">depth_bigwig_file</span>,</span>
<span>  <span class="va">variant_file</span>,</span>
<span>  <span class="va">txdb</span>,</span>
<span>  <span class="va">org</span>,</span>
<span>  gene_delta_threshold <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  downsample <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  line_size <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  line_color <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>  colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  max_value <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min_value <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min_variant_distance <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fit"</span>, <span class="st">"delta"</span>, <span class="st">"loess"</span><span class="op">)</span>,</span>
<span>  trend_window <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  apply_weight <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  outside_weight <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  inside_weight <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  variant_alpha <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  nudge_y <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  samplename <span class="op">=</span> <span class="st">""</span>,</span>
<span>  chr_filter <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  exclude_xy <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  highlight_genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gband_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"~/develop/pacbiowdlR/cytobands.tsv"</span><span class="op">)</span>,</span>
<span>  gband_y_offset <span class="op">=</span> <span class="op">-</span><span class="fl">0.2</span>,</span>
<span>  gband_text_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  showCNAbands <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  trend_regions <span class="op">=</span> <span class="st">"both"</span>,</span>
<span>  return_data <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-depth-bigwig-file">depth_bigwig_file<a class="anchor" aria-label="anchor" href="#arg-depth-bigwig-file"></a></dt>
<dd><p>Character. File path to a BigWig file containing coverage depth data.</p></dd>


<dt id="arg-variant-file">variant_file<a class="anchor" aria-label="anchor" href="#arg-variant-file"></a></dt>
<dd><p>Character. File path to a VCF file with variant data.</p></dd>


<dt id="arg-txdb">txdb<a class="anchor" aria-label="anchor" href="#arg-txdb"></a></dt>
<dd><p>A transcript database object used for gene annotation.</p></dd>


<dt id="arg-org">org<a class="anchor" aria-label="anchor" href="#arg-org"></a></dt>
<dd><p>A species annotation object (e.g., <code>org.Hs.eg.db</code>) for mapping gene identifiers.</p></dd>


<dt id="arg-gene-delta-threshold">gene_delta_threshold<a class="anchor" aria-label="anchor" href="#arg-gene-delta-threshold"></a></dt>
<dd><p>Numeric. Delta threshold for gene annotation filtering.</p></dd>


<dt id="arg-downsample">downsample<a class="anchor" aria-label="anchor" href="#arg-downsample"></a></dt>
<dd><p>Numeric. Proportion of the coverage data to retain after downsampling.</p></dd>


<dt id="arg-point-size">point_size<a class="anchor" aria-label="anchor" href="#arg-point-size"></a></dt>
<dd><p>Numeric. Size of individual points in the plot.</p></dd>


<dt id="arg-line-size">line_size<a class="anchor" aria-label="anchor" href="#arg-line-size"></a></dt>
<dd><p>Numeric. Size of the trend line.</p></dd>


<dt id="arg-line-color">line_color<a class="anchor" aria-label="anchor" href="#arg-line-color"></a></dt>
<dd><p>Character. Color of the trend line.</p></dd>


<dt id="arg-colors">colors<a class="anchor" aria-label="anchor" href="#arg-colors"></a></dt>
<dd><p>Named vector of colors for chromosomes; if <code>NULL</code>, defaults to an alternating palette.</p></dd>


<dt id="arg-max-value">max_value<a class="anchor" aria-label="anchor" href="#arg-max-value"></a></dt>
<dd><p>Numeric. Maximum allowed delta value; values above this are capped.</p></dd>


<dt id="arg-min-value">min_value<a class="anchor" aria-label="anchor" href="#arg-min-value"></a></dt>
<dd><p>Numeric. Minimum allowed delta value; values below this are capped.</p></dd>


<dt id="arg-min-variant-distance">min_variant_distance<a class="anchor" aria-label="anchor" href="#arg-min-variant-distance"></a></dt>
<dd><p>Numeric. Minimum variant length (in bp) to include.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>Character. One of <code>"fit"</code>, <code>"delta"</code>, or <code>"loess"</code> determining the method used to compute delta values.</p></dd>


<dt id="arg-trend-window">trend_window<a class="anchor" aria-label="anchor" href="#arg-trend-window"></a></dt>
<dd><p>Integer. Window size for computing the running median trend line.</p></dd>


<dt id="arg-apply-weight">apply_weight<a class="anchor" aria-label="anchor" href="#arg-apply-weight"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, applies a weight multiplier to delta values outside CNA calls.</p></dd>


<dt id="arg-outside-weight">outside_weight<a class="anchor" aria-label="anchor" href="#arg-outside-weight"></a></dt>
<dd><p>Numeric. Weight multiplier applied to delta values outside CNA calls.</p></dd>


<dt id="arg-inside-weight">inside_weight<a class="anchor" aria-label="anchor" href="#arg-inside-weight"></a></dt>
<dd><p>Numeric. Weight multiplier applied to delta values inside CNA calls.</p></dd>


<dt id="arg-variant-alpha">variant_alpha<a class="anchor" aria-label="anchor" href="#arg-variant-alpha"></a></dt>
<dd><p>Numeric. Transparency level for variant call rectangles.</p></dd>


<dt id="arg-nudge-y">nudge_y<a class="anchor" aria-label="anchor" href="#arg-nudge-y"></a></dt>
<dd><p>Numeric. Vertical nudge for positioning gene labels.</p></dd>


<dt id="arg-samplename">samplename<a class="anchor" aria-label="anchor" href="#arg-samplename"></a></dt>
<dd><p>Character. Sample label used in the plot title.</p></dd>


<dt id="arg-chr-filter">chr_filter<a class="anchor" aria-label="anchor" href="#arg-chr-filter"></a></dt>
<dd><p>Character. If provided, restricts the analysis to the specified chromosome.</p></dd>


<dt id="arg-exclude-xy">exclude_xy<a class="anchor" aria-label="anchor" href="#arg-exclude-xy"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, chromosomes X and Y are excluded.</p></dd>


<dt id="arg-highlight-genes">highlight_genes<a class="anchor" aria-label="anchor" href="#arg-highlight-genes"></a></dt>
<dd><p>Character vector. Specific gene symbols to highlight in the plot.</p></dd>


<dt id="arg-gband-file">gband_file<a class="anchor" aria-label="anchor" href="#arg-gband-file"></a></dt>
<dd><p>Character. File path to a cytogenetic band data file (e.g., a TSV file).</p></dd>


<dt id="arg-gband-y-offset">gband_y_offset<a class="anchor" aria-label="anchor" href="#arg-gband-y-offset"></a></dt>
<dd><p>Numeric. Vertical offset for G-band label placement.</p></dd>


<dt id="arg-gband-text-size">gband_text_size<a class="anchor" aria-label="anchor" href="#arg-gband-text-size"></a></dt>
<dd><p>Numeric. Text size for G-band labels.</p></dd>


<dt id="arg-showcnabands">showCNAbands<a class="anchor" aria-label="anchor" href="#arg-showcnabands"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, displays G-band annotations on the plot.</p></dd>


<dt id="arg-trend-regions">trend_regions<a class="anchor" aria-label="anchor" href="#arg-trend-regions"></a></dt>
<dd><p>Character. One of <code>"both"</code>, <code>"inside"</code>, or <code>"outside"</code>.
This option controls where the running median trend line is displayed:</p><dl><dt><code>"both"</code></dt>
<dd><p>Trend line is plotted for all points (default).</p></dd>

  <dt><code>"inside"</code></dt>
<dd><p>Trend line is only plotted for positions that fall within variant regions.</p></dd>

  <dt><code>"outside"</code></dt>
<dd><p>Trend line is only plotted for positions that fall outside variant regions.</p></dd>


</dl></dd>


<dt id="arg-return-data">return_data<a class="anchor" aria-label="anchor" href="#arg-return-data"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, returns a list containing the plot, the coverage data, variant calls, and gene annotations.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Either a <code>ggplot2</code> object representing the CNA plot with gene highlights, or a list with additional data when <code>return_data</code> is <code>TRUE</code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The function performs several steps:</p><ol><li><p>Imports and filters coverage data from the BigWig file using <code>import.bw()</code> and retains standard chromosomes (using <code>keepStandardChromosomes()</code>).</p></li>
<li><p>Optionally loads external GC/repeat data when using the <code>"fit"</code> method.</p></li>
<li><p>Downsamples the coverage data.</p></li>
<li><p>Computes delta values using the specified method:</p><ul><li><p>[<code>"delta"</code>] subtracts the mean coverage.</p></li>
<li><p>[<code>"loess"</code>] fits a LOESS model and computes log2 ratios.</p></li>
<li><p>[<code>"fit"</code>] fits a linear model to predict coverage based on GC content and repeat fraction.</p></li>
</ul></li>
<li><p>Reads and filters variant calls from the VCF file.</p></li>
<li><p>Computes genomic offsets via <code><a href="generate_offsets.html">generate_offsets()</a></code> and marks variant regions.</p></li>
<li><p>Applies weighting to delta values with different multipliers for points inside and outside CNV calls.</p></li>
<li><p>Computes a running median trend line using <code>rollapply()</code>, and, based on <code>trend_regions</code>,
        subsets the trend line to display only points inside variants, outside variants, or in both regions.</p></li>
<li><p>Annotates gene information on the coverage data using <code><a href="annotateCoverageWithGenes.html">annotateCoverageWithGenes()</a></code>.</p></li>
<li><p>Constructs the final plot with <code>ggplot2</code> showing data points, the trend line, variant rectangles, and gene labels (with optional G-band annotations).</p></li>
</ol></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Scott Furlan.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>


<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({


    apiKey: 'YOUR_API_KEY',
    indexName: 'pacbiowdlR',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script></body></html>

