<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate CNA Plot from Regions — generateCNAPlotFromRegions2 • pacbiowdlR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous"><link href="../docsearch.css" rel="stylesheet"><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Generate CNA Plot from Regions — generateCNAPlotFromRegions2"><meta property="og:description" content="Generates a genome-wide copy number alteration (CNA) plot by combining coverage data from a BigWig
file and variant calls from a VCF file. Delta values are computed from the coverage data and overlaid
with variant regions. Optionally, cytoband annotations can be added."><meta property="og:image" content="https://scfurl.github.io/pacbiowdlR/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pacbiowdlR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>

  </a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/scfurl/pacbiowdlR" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
<li>
  <a href="https://bsky.app/profile/scfurl.bsky.social" class="external-link">
    <span class="fa fa-bluesky"></span>

  </a>
</li>
      </ul><form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off"></div>
      </form>

    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate CNA Plot from Regions</h1>

    <div class="hidden name"><code>generateCNAPlotFromRegions2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generates a genome-wide copy number alteration (CNA) plot by combining coverage data from a BigWig
file and variant calls from a VCF file. Delta values are computed from the coverage data and overlaid
with variant regions. Optionally, cytoband annotations can be added.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">generateCNAPlotFromRegions2</span><span class="op">(</span></span>
<span>  <span class="va">depth_bigwig_file</span>,</span>
<span>  <span class="va">variant_file</span>,</span>
<span>  threshold <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  downsample <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  log_scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  remove_centromeres <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  max_value <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min_value <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min_variant_distance <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  samplename <span class="op">=</span> <span class="st">""</span>,</span>
<span>  show_cytobands <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cytoband_angle <span class="op">=</span> <span class="fl">90</span>,</span>
<span>  chr_filter <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  return_data <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-depth-bigwig-file">depth_bigwig_file<a class="anchor" aria-label="anchor" href="#arg-depth-bigwig-file"></a></dt>
<dd><p>Character. File path to a BigWig file containing coverage (depth) information.</p></dd>


<dt id="arg-variant-file">variant_file<a class="anchor" aria-label="anchor" href="#arg-variant-file"></a></dt>
<dd><p>Character. File path to a VCF file containing variant call information.</p></dd>


<dt id="arg-threshold">threshold<a class="anchor" aria-label="anchor" href="#arg-threshold"></a></dt>
<dd><p>Numeric. Delta threshold used for display. Default is 1.5.</p></dd>


<dt id="arg-downsample">downsample<a class="anchor" aria-label="anchor" href="#arg-downsample"></a></dt>
<dd><p>Numeric. Proportion of the coverage data to retain after downsampling. Default is 0.01.</p></dd>


<dt id="arg-point-size">point_size<a class="anchor" aria-label="anchor" href="#arg-point-size"></a></dt>
<dd><p>Numeric. Size of the points in the plot. Default is 0.5.</p></dd>


<dt id="arg-colors">colors<a class="anchor" aria-label="anchor" href="#arg-colors"></a></dt>
<dd><p>Named vector. Colors to be used for chromosomes; if <code>NULL</code>, default colors are assigned.</p></dd>


<dt id="arg-log-scale">log_scale<a class="anchor" aria-label="anchor" href="#arg-log-scale"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, applies a log scale (unused in code, but available for future extension). Default is <code>TRUE</code>.</p></dd>


<dt id="arg-remove-centromeres">remove_centromeres<a class="anchor" aria-label="anchor" href="#arg-remove-centromeres"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, centromere regions are removed (unused in code, but available for extension). Default is <code>TRUE</code>.</p></dd>


<dt id="arg-max-value">max_value<a class="anchor" aria-label="anchor" href="#arg-max-value"></a></dt>
<dd><p>Numeric. Maximum delta value for setting plot y-axis limits. Default is <code>NULL</code>.</p></dd>


<dt id="arg-min-value">min_value<a class="anchor" aria-label="anchor" href="#arg-min-value"></a></dt>
<dd><p>Numeric. Minimum delta value for setting plot y-axis limits. Default is <code>NULL</code>.</p></dd>


<dt id="arg-min-variant-distance">min_variant_distance<a class="anchor" aria-label="anchor" href="#arg-min-variant-distance"></a></dt>
<dd><p>Numeric. Minimum distance (in bp) for a variant call to be retained. Default is 10000.</p></dd>


<dt id="arg-samplename">samplename<a class="anchor" aria-label="anchor" href="#arg-samplename"></a></dt>
<dd><p>Character. A label for the sample which is displayed in the plot title.</p></dd>


<dt id="arg-show-cytobands">show_cytobands<a class="anchor" aria-label="anchor" href="#arg-show-cytobands"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, cytoband labels are added to the variant calls. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-cytoband-angle">cytoband_angle<a class="anchor" aria-label="anchor" href="#arg-cytoband-angle"></a></dt>
<dd><p>Numeric. Angle (in degrees) to rotate the cytoband text labels. Default is 90.</p></dd>


<dt id="arg-chr-filter">chr_filter<a class="anchor" aria-label="anchor" href="#arg-chr-filter"></a></dt>
<dd><p>Character. If provided, limits the analysis to the specified chromosome.</p></dd>


<dt id="arg-return-data">return_data<a class="anchor" aria-label="anchor" href="#arg-return-data"></a></dt>
<dd><p>Logical. If <code>TRUE</code>, returns a list containing the plot along with the underlying
coverage and variant call data. Default is <code>FALSE</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Either a <code>ggplot2</code> object representing the CNA plot, or, if <code>return_data</code> is <code>TRUE</code>,
  a list with elements <code>plot</code>, <code>coverage_data</code>, and <code>variant_calls</code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function performs the following steps:</p><ol><li><p>Imports coverage data from a BigWig file using <code>import.bw()</code> from <code>rtracklayer</code>
          and converts it into a data frame.</p></li>
<li><p>Filters for valid chromosomes (chr1-22, X, Y), optionally restricting to a single chromosome.</p></li>
<li><p>Downsamples the coverage data and computes a delta value for each data point.</p></li>
<li><p>Reads variant calls from a VCF file using <code>fread()</code> from <code>data.table</code>, parsing out
          the end positions and variant types.</p></li>
<li><p>Optionally annotates cytoband information for each variant call using overlaps with
          <code>hg19IdeogramCyto</code> (assumed to be available in the global environment).</p></li>
<li><p>Filters variant calls by a minimum distance if specified.</p></li>
<li><p>Computes cumulative genomic offsets via the <code><a href="generate_offsets.html">generate_offsets()</a></code> function.</p></li>
<li><p>Constructs and returns a plot built with <code>ggplot2</code> that shows the coverage data points,
          highlighted variant regions, and optionally cytoband labels.</p></li>
</ol></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Scott Furlan.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>


<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({


    apiKey: 'YOUR_API_KEY',
    indexName: 'pacbiowdlR',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script></body></html>

