<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Plotting DNA Copy Number Alterations • pacbiowdlR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Plotting DNA Copy Number Alterations">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pacbiowdlR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>

  </a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/scfurl/pacbiowdlR" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
<li>
  <a href="https://bsky.app/profile/scfurl.bsky.social" class="external-link">
    <span class="fa fa-bluesky"></span>

  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>

    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Plotting DNA Copy Number Alterations</h1>
                        <h4 data-toc-skip class="author">Scott
Furlan</h4>
            
            <h4 data-toc-skip class="date">2025-04-15</h4>
      

      <div class="hidden name"><code>CNAplot.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="theory">Theory<a class="anchor" aria-label="anchor" href="#theory"></a>
</h2>
<p>To make a CNA plot in <em>pacbiowdlR</em>, you need two files as
input both of which are output in the <em>pacbiowdlR</em> pipeline: 1) a
depth file output and 2) a variant file which denotes the copy
alteration ranges.</p>
<p>There are three different methods used to visualize the delta values
in your CNA plot. In each case, let</p>
<ul>
<li>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>s</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">s(x)</annotation></semantics></math>
denote the observed score (or coverage value) at position
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>x</mi><annotation encoding="application/x-tex">x</annotation></semantics></math>,</li>
<li>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>δ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\delta(x)</annotation></semantics></math>
be the computed delta value at position
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>x</mi><annotation encoding="application/x-tex">x</annotation></semantics></math>.</li>
</ul>
<div class="section level3">
<h3 id="simple-difference-delta-method">1. Simple Difference (“delta” method)<a class="anchor" aria-label="anchor" href="#simple-difference-delta-method"></a>
</h3>
<p>In the simplest method, the delta is computed as the difference
between the observed score and the mean score over all positions.
Mathematically, if the mean score is</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mover><mi>s</mi><mo accent="true">‾</mo></mover><mo>=</mo><mfrac><mn>1</mn><mi>N</mi></mfrac><munderover><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mi>N</mi></munderover><mi>s</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mi>i</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>,</mo></mrow><annotation encoding="application/x-tex">
\bar{s} = \frac{1}{N} \sum_{i=1}^{N} s(x_i),
</annotation></semantics></math></p>
<p>then the delta for a given position
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>x</mi><annotation encoding="application/x-tex">x</annotation></semantics></math>
is</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>δ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mi>s</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>−</mo><mover><mi>s</mi><mo accent="true">‾</mo></mover><mi>.</mi></mrow><annotation encoding="application/x-tex">
\delta(x) = s(x) - \bar{s}.
</annotation></semantics></math></p>
<p>This method highlights how much a given data point deviates from the
overall average.</p>
<hr>
</div>
<div class="section level3">
<h3 id="loess-based-log-ratio-loess-method">2. LOESS-Based Log-Ratio (“loess” method)<a class="anchor" aria-label="anchor" href="#loess-based-log-ratio-loess-method"></a>
</h3>
<p>In this approach, a nonparametric regression (LOESS) is used to fit a
smooth curve to the observed scores as a function of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>x</mi><annotation encoding="application/x-tex">x</annotation></semantics></math>.
Let the LOESS-fitted values be denoted by
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mover><mi>s</mi><mo accent="true">̂</mo></mover><mtext mathvariant="normal">LOESS</mtext></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\hat{s}_{\text{LOESS}}(x)</annotation></semantics></math>.
Then, the delta is computed as the log<sub>2</sub> ratio of the observed
score to the fitted value:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>δ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><msub><mo>log</mo><mn>2</mn></msub><mrow><mo stretchy="true" form="prefix">(</mo><mfrac><mrow><mi>s</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><msub><mover><mi>s</mi><mo accent="true">̂</mo></mover><mtext mathvariant="normal">LOESS</mtext></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac><mo stretchy="true" form="postfix">)</mo></mrow><mi>.</mi></mrow><annotation encoding="application/x-tex">
\delta(x) = \log_2\left(\frac{s(x)}{\hat{s}_{\text{LOESS}}(x)}\right).
</annotation></semantics></math></p>
<p>This transformation emphasizes relative differences rather than
absolute differences. Using the log ratio, you capture percentage-like
changes in coverage relative to the locally expected value.</p>
<hr>
</div>
<div class="section level3">
<h3 id="linear-model-fit-based-log-ratio-fit-method-preferred">3. Linear Model Fit-Based Log-Ratio (“fit” method) – PREFERRED<a class="anchor" aria-label="anchor" href="#linear-model-fit-based-log-ratio-fit-method-preferred"></a>
</h3>
<p>For the “fit” method, a linear regression model is used to predict
the coverage score based on predictors such as GC content and repeat
fraction. Suppose the model is given by</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>s</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><msub><mi>β</mi><mn>0</mn></msub><mo>+</mo><msub><mi>β</mi><mn>1</mn></msub><mo>⋅</mo><mtext mathvariant="normal">gc_content</mtext><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>β</mi><mn>2</mn></msub><mo>⋅</mo><mtext mathvariant="normal">repeat_fraction</mtext><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mi>ε</mi><mo>,</mo></mrow><annotation encoding="application/x-tex">
s(x) = \beta_0 + \beta_1 \cdot \text{gc\_content}(x) + \beta_2 \cdot \text{repeat\_fraction}(x) + \varepsilon,
</annotation></semantics></math></p>
<p>and let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mover><mi>s</mi><mo accent="true">̂</mo></mover><mtext mathvariant="normal">fit</mtext></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\hat{s}_{\text{fit}}(x)</annotation></semantics></math>
be the predicted (fitted) value from this linear model. Then, the delta
is defined in the same way as for the LOESS method but using the fitted
score:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>δ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><msub><mo>log</mo><mn>2</mn></msub><mrow><mo stretchy="true" form="prefix">(</mo><mfrac><mrow><mi>s</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><msub><mover><mi>s</mi><mo accent="true">̂</mo></mover><mtext mathvariant="normal">fit</mtext></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>x</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac><mo stretchy="true" form="postfix">)</mo></mrow><mi>.</mi></mrow><annotation encoding="application/x-tex">
\delta(x) = \log_2\left(\frac{s(x)}{\hat{s}_{\text{fit}}(x)}\right).
</annotation></semantics></math></p>
<p>Here, the log<sub>2</sub> ratio provides a measure of how the
observed score deviates from what is expected based on the linear
relationship with GC content and repeat fraction.</p>
<p>Each method has its own advantage: the simple difference is easy to
compute and interpret, while the log-ratio methods (“loess” and “fit”)
provide a relative measure that can stabilize variance and better
highlight proportional changes across the genome. The “loess” method is
significantly slower, so we plot the other two here. The delta method
produces a wider range of values, so we recommend not constraining the
plot dimensions. Otherwise the dimensions can be specified as below.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scfurl.github.io/pacbiowdlR/">pacbiowdlR</a></span><span class="op">)</span></span>
<span><span class="va">depth_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"examples/K562.GRCh38.hificnv.depth.bw"</span>, package <span class="op">=</span> <span class="st">"pacbiowdlR"</span><span class="op">)</span></span>
<span><span class="va">cna_calls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"examples/K562.GRCh38.hificnv.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"pacbiowdlR"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/CNAPlot.html">CNAPlot</a></span><span class="op">(</span>depth_bigwig_file  <span class="op">=</span>   <span class="va">depth_file</span>,</span>
<span>  variant_file       <span class="op">=</span>   <span class="va">cna_calls</span>,</span>
<span>  downsample         <span class="op">=</span> <span class="fl">0.1</span>,  method <span class="op">=</span> <span class="st">"delta"</span>,</span>
<span>  samplename         <span class="op">=</span> <span class="st">"K562"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CNAplot_files/figure-html/unnamed-chunk-2-1.png" width="120%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/CNAPlot.html">CNAPlot</a></span><span class="op">(</span>depth_bigwig_file  <span class="op">=</span>   <span class="va">depth_file</span>,</span>
<span>  variant_file       <span class="op">=</span>   <span class="va">cna_calls</span>, method <span class="op">=</span> <span class="st">"fit"</span>, </span>
<span>  samplename         <span class="op">=</span> <span class="st">"K562"</span>, max_value <span class="op">=</span> <span class="fl">3</span>, min_value <span class="op">=</span> <span class="op">-</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="CNAplot_files/figure-html/unnamed-chunk-2-2.png" width="120%" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="some-plotting-aesthetics">Some plotting aesthetics<a class="anchor" aria-label="anchor" href="#some-plotting-aesthetics"></a>
</h2>
<p>In the below we define some colors and replot. Using these colors a
black-colored fit line looks best</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chrom_palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"#FF0000"</span>, <span class="st">"#FF9900"</span>, <span class="st">"#FFCC00"</span>, <span class="st">"#00FF00"</span>, <span class="st">"#6699FF"</span>, <span class="st">"#CC33FF"</span>,</span>
<span>  <span class="st">"#999912"</span>, <span class="st">"#999999"</span>, <span class="st">"#FF00CC"</span>, <span class="st">"#CC0000"</span>, <span class="st">"#FFCCCC"</span>, <span class="st">"#FFFF00"</span>,</span>
<span>  <span class="st">"#CCFF00"</span>, <span class="st">"#358000"</span>, <span class="st">"#0000CC"</span>, <span class="st">"#99CCFF"</span>, <span class="st">"#00FFFF"</span>, <span class="st">"#ECFFFF"</span>,</span>
<span>  <span class="st">"#9900CC"</span>, <span class="st">"#CC99FF"</span>, <span class="st">"#996600"</span>, <span class="st">"#666600"</span>, <span class="st">"#666666"</span>, <span class="st">"#CCCCCC"</span>,</span>
<span>  <span class="st">"#79CC3B"</span>, <span class="st">"#E0EC3B"</span>, <span class="st">"#CCC99F"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/CNAPlot.html">CNAPlot</a></span><span class="op">(</span>depth_bigwig_file  <span class="op">=</span>   <span class="va">depth_file</span>,</span>
<span>  variant_file       <span class="op">=</span>   <span class="va">cna_calls</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"fit"</span>, line_color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  samplename         <span class="op">=</span> <span class="st">"K562"</span>, max_value <span class="op">=</span> <span class="fl">3</span>, min_value <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, colors <span class="op">=</span> <span class="va">chrom_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="CNAplot_files/figure-html/unnamed-chunk-3-1.png" width="120%" style="display: block; margin: auto;"></p>
<p>By default, CNAPlot will “accentuate” copy number changes in deleted
or duplicated regions by applying a weight to those regions outside of
variants. This effectively minimimizes the visualized variance in these
regions. This feature can be turned off, or modulated by adjusting the
weight via the <code>outside_weight</code> parameter. One can also
change the weighting of the <code>inside_weight</code> to further
maximize visualization. However it is critical to note that these
weights are applied AFTER estimating the y-axis values, therefore one
should interpret the values with caution.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/CNAPlot.html">CNAPlot</a></span><span class="op">(</span>depth_bigwig_file  <span class="op">=</span>   <span class="va">depth_file</span>,</span>
<span>  variant_file       <span class="op">=</span>   <span class="va">cna_calls</span>, outside_weight <span class="op">=</span> <span class="fl">0.4</span>, inside_weight <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"fit"</span>, line_color <span class="op">=</span> <span class="st">"black"</span>, max_value <span class="op">=</span> <span class="fl">5</span>, min_value <span class="op">=</span> <span class="op">-</span><span class="fl">5</span>,</span>
<span>  samplename         <span class="op">=</span> <span class="st">"K562"</span>, colors <span class="op">=</span> <span class="va">chrom_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="CNAplot_files/figure-html/unnamed-chunk-4-1.png" width="150%" style="display: block; margin: auto;"></p>
<div class="section level3">
<h3 id="deeper-analysis-of-cna">Deeper analysis of CNA<a class="anchor" aria-label="anchor" href="#deeper-analysis-of-cna"></a>
</h3>
<p>Using <em>pacbiowdlR</em>, one can take a deeper look at copy number
alterations. For example in Chr9 of the K562 data, we see regions of
signifiantly deleted DNA (as evidenced by coverage inference). We can
use the <code>CNAPlot_Highlight</code> function to discover genes
amplified in this region. Two additional objects are needed for gene
highlighting, a txdb and org annotation file.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/CNAPlot_Highlight.html">CNAPlot_Highlight</a></span><span class="op">(</span>depth_bigwig_file  <span class="op">=</span>   <span class="va">depth_file</span>, </span>
<span>                  txdb <span class="op">=</span> <span class="fu">TxDb.Hsapiens.UCSC.hg38.knownGene</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/TxDb.Hsapiens.UCSC.hg38.knownGene/man/package.html" class="external-link">TxDb.Hsapiens.UCSC.hg38.knownGene</a></span>, </span>
<span>                  org <span class="op">=</span> <span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.egBASE.html" class="external-link">org.Hs.eg.db</a></span>, gene_delta_threshold <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>                  variant_file       <span class="op">=</span>   <span class="va">cna_calls</span>, </span>
<span>                  method <span class="op">=</span> <span class="st">"fit"</span>, line_color <span class="op">=</span> <span class="st">"black"</span>, chr_filter <span class="op">=</span> <span class="st">"chr9"</span>,</span>
<span>                  samplename         <span class="op">=</span> <span class="st">"K562"</span>, colors <span class="op">=</span> <span class="va">chrom_palette</span><span class="op">)</span></span></code></pre></div>
<p>One will not that there are regions that plot as having a 0 value for
DNA content. This results when the log2 ratio is infinitely negative and
are indicative of a complete loss of DNA in the region. This can be
confirmed using the <code>delta</code> method of visualization as shown
below. We can increase the number of events plotted using the
<code>downsample</code> parameter. Unsurprisingly the region lost
contains the tumor suppressors CDKN2A, CDKN2B.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/CNAPlot_Highlight.html">CNAPlot_Highlight</a></span><span class="op">(</span>depth_bigwig_file  <span class="op">=</span>   <span class="va">depth_file</span>, </span>
<span>                  txdb <span class="op">=</span> <span class="fu">TxDb.Hsapiens.UCSC.hg38.knownGene</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/TxDb.Hsapiens.UCSC.hg38.knownGene/man/package.html" class="external-link">TxDb.Hsapiens.UCSC.hg38.knownGene</a></span>, </span>
<span>                  org <span class="op">=</span> <span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.egBASE.html" class="external-link">org.Hs.eg.db</a></span>, downsample <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                  variant_file       <span class="op">=</span>   <span class="va">cna_calls</span>, max_value <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                  method <span class="op">=</span> <span class="st">"delta"</span>, line_color <span class="op">=</span> <span class="st">"black"</span>, chr_filter <span class="op">=</span> <span class="st">"chr9"</span>, highlight_genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CDKN2A"</span>, <span class="st">"CDKN2B"</span><span class="op">)</span>, </span>
<span>                  samplename         <span class="op">=</span> <span class="st">"K562"</span>, colors <span class="op">=</span> <span class="va">chrom_palette</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Scott Furlan.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>


<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({


    apiKey: 'YOUR_API_KEY',
    indexName: 'pacbiowdlR',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
