<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interfacing with the PacBioWDL pipeline output â€¢ pacbiowdlR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Interfacing with the PacBioWDL pipeline output">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pacbiowdlR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>

  </a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/scfurl/pacbiowdlR" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
<li>
  <a href="https://bsky.app/profile/scfurl.bsky.social" class="external-link">
    <span class="fa fa-bluesky"></span>

  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>

    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Interfacing with the PacBioWDL pipeline
output</h1>
                        <h4 data-toc-skip class="author">Scott
Furlan</h4>
            
            <h4 data-toc-skip class="date">2025-04-16</h4>
      

      <div class="hidden name"><code>Interface.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="pacbio-wdl-pipeline-json-handling-guide">PacBio WDL Pipeline JSON Handling Guide<a class="anchor" aria-label="anchor" href="#pacbio-wdl-pipeline-json-handling-guide"></a>
</h2>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>The PacBio WDL (Workflow Description Language) pipeline generates a
wealth of output files organized in a structured hierarchy. This guide
explains how to use the <code><a href="../reference/read_json_file.html">read_json_file()</a></code> function to easily
access these outputs across different computing environments.</p>
</div>
<div class="section level3">
<h3 id="the-path-translation-problem">The Path Translation Problem<a class="anchor" aria-label="anchor" href="#the-path-translation-problem"></a>
</h3>
<p>When working with genomic pipelines in collaborative environments,
file paths often need translation between: - Server paths (e.g.,
<code>/server/share/...</code>) - Local mounted drives (e.g.,
<code>/Volumes/...</code>)</p>
<p>This creates challenges when configuration files contain absolute
paths that need to be accessed from different systems.</p>
</div>
<div class="section level3">
<h3 id="using-read_json_file-function">Using <code>read_json_file()</code> Function<a class="anchor" aria-label="anchor" href="#using-read_json_file-function"></a>
</h3>
<p>The <code><a href="../reference/read_json_file.html">read_json_file()</a></code> function solves this problem by
reading JSON configuration files and automatically translating paths
between environments.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">read_json_file</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">file_path</span>, <span class="va">sub1</span> <span class="op">=</span> <span class="st">"/fh/working"</span>, <span class="va">sub2</span> <span class="op">=</span> <span class="st">"/Volumes"</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Check if file exists</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">file_path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"File does not exist: "</span>, <span class="va">file_path</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="co"># Read and parse JSON</span></span>
<span>  <span class="va">json_data</span> <span class="op">&lt;-</span> <span class="fu">fromJSON</span><span class="op">(</span><span class="va">file_path</span><span class="op">)</span></span>
<span>  <span class="va">json_data_e</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">json_data</span>, <span class="kw">function</span><span class="op">(</span><span class="va">loc</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="va">sub1</span>, <span class="va">sub2</span>, <span class="va">loc</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">json_data_e</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="workflow-example">Workflow Example<a class="anchor" aria-label="anchor" href="#workflow-example"></a>
</h3>
<div class="section level4">
<h4 id="load-the-output-configuration">1. Load the Output Configuration<a class="anchor" aria-label="anchor" href="#load-the-output-configuration"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load output configuration from PacBio WDL pipeline</span></span>
<span><span class="va">data_K562</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_json_file.html">read_json_file</a></span><span class="op">(</span><span class="st">"pacbio_outputs.json"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="access-specific-output-files">2. Access Specific Output Files<a class="anchor" aria-label="anchor" href="#access-specific-output-files"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Access structural variant VCF file</span></span>
<span><span class="va">sv_vcf</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.phased_sv_vcf</span></span>
<span><span class="co"># [1] "/Volumes/furlan_s/sfurlan/pacbiorerun/pbWGS/K562/20250211_115553_humanwgs_singleton/out/phased_sv_vcf/K562.GRCh38.structural_variants.phased.vcf.gz"</span></span>
<span></span>
<span><span class="co"># Access small variant statistics</span></span>
<span><span class="va">small_var_stats</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.small_variant_stats</span></span>
<span><span class="co"># [1] "/Volumes/furlan_s/sfurlan/pacbiorerun/pbWGS/K562/20250211_115553_humanwgs_singleton/out/small_variant_stats/K562.GRCh38.small_variants.vcf.stats.txt"</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="available-output-types">Available Output Types<a class="anchor" aria-label="anchor" href="#available-output-types"></a>
</h3>
<p>The PacBio WDL pipeline produces numerous output files organized by
category:</p>
<div class="section level4">
<h4 id="key-file-categories">Key File Categories<a class="anchor" aria-label="anchor" href="#key-file-categories"></a>
</h4>
<table style="width:100%;" class="table">
<colgroup>
<col width="25%">
<col width="33%">
<col width="41%">
</colgroup>
<thead><tr class="header">
<th>Category</th>
<th>Description</th>
<th>Example Output</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Aligned Reads</strong></td>
<td>BAM files with mapped reads</td>
<td><code>humanwgs_singleton.merged_haplotagged_bam</code></td>
</tr>
<tr class="even">
<td><strong>Small Variants</strong></td>
<td>SNVs and small indels</td>
<td><code>humanwgs_singleton.phased_small_variant_vcf</code></td>
</tr>
<tr class="odd">
<td><strong>Structural Variants</strong></td>
<td>Large genomic rearrangements</td>
<td><code>humanwgs_singleton.phased_sv_vcf</code></td>
</tr>
<tr class="even">
<td><strong>Copy Number Variants</strong></td>
<td>Deletions and duplications</td>
<td><code>humanwgs_singleton.cnv_vcf</code></td>
</tr>
<tr class="odd">
<td><strong>Methylation Data</strong></td>
<td>CpG methylation patterns</td>
<td><code>humanwgs_singleton.cpg_combined_bed</code></td>
</tr>
<tr class="even">
<td><strong>Tandem Repeats</strong></td>
<td>Repeat expansions</td>
<td><code>humanwgs_singleton.phased_trgt_vcf</code></td>
</tr>
<tr class="odd">
<td><strong>Phasing Information</strong></td>
<td>Haplotype assignments</td>
<td><code>humanwgs_singleton.phase_blocks</code></td>
</tr>
<tr class="even">
<td><strong>Quality Metrics</strong></td>
<td>Various QC statistics</td>
<td><code>humanwgs_singleton.read_length_plot</code></td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="key-statistics">Key Statistics<a class="anchor" aria-label="anchor" href="#key-statistics"></a>
</h4>
<p>The JSON also contains pre-computed statistics:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Basic sequencing stats</span></span>
<span><span class="va">total_reads</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.stat_num_reads</span>                <span class="co"># [1] "5039572"</span></span>
<span><span class="va">mapped_percent</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.stat_mapped_percent</span>        <span class="co"># [1] "99.98"</span></span>
<span><span class="va">mean_depth</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.stat_mean_depth</span>                <span class="co"># [1] "33.38"</span></span>
<span></span>
<span><span class="co"># Variant statistics  </span></span>
<span><span class="va">snv_count</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.stat_small_variant_SNV_count</span>    <span class="co"># [1] "3535236"</span></span>
<span><span class="va">indel_count</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.stat_small_variant_INDEL_count</span> <span class="co"># [1] "701026"</span></span>
<span><span class="va">sv_del_count</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.stat_sv_DEL_count</span>           <span class="co"># [1] "9106"</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="tips-for-working-with-pacbio-wdl-outputs">Tips for Working with PacBio WDL Outputs<a class="anchor" aria-label="anchor" href="#tips-for-working-with-pacbio-wdl-outputs"></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<p><strong>Custom Path Mapping</strong>: Customize the
<code>sub1</code> and <code>sub2</code> parameters based on your
specific environment</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For different mount points</span></span>
<span><span class="va">data_K562</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_json_file.html">read_json_file</a></span><span class="op">(</span><span class="st">"pacbio_outputs.json"</span>, sub1 <span class="op">=</span> <span class="st">"/data/projects"</span>, sub2 <span class="op">=</span> <span class="st">"/mnt/data"</span><span class="op">)</span></span></code></pre></div>
</li>
<li>
<p><strong>File Existence Check</strong>: Always verify file
existence before attempting to read</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sv_vcf_path</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.phased_sv_vcf</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">sv_vcf_path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Process file</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/warning.html" class="external-link">warning</a></span><span class="op">(</span><span class="st">"SV VCF file not found at expected location"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</li>
<li>
<p><strong>Working with Tabix-Indexed Files</strong>: Many output
files are compressed and indexed</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Rsamtools" class="external-link">Rsamtools</a></span><span class="op">)</span></span>
<span><span class="co"># Access indexed VCF</span></span>
<span><span class="va">vcf</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.phased_small_variant_vcf</span></span>
<span><span class="va">vcf_index</span> <span class="op">&lt;-</span> <span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.phased_small_variant_vcf_index</span></span>
<span><span class="co"># Use with appropriate packages for indexed access</span></span></code></pre></div>
</li>
</ol>
</div>
<div class="section level3">
<h3 id="example-analysis-workflow">Example Analysis Workflow<a class="anchor" aria-label="anchor" href="#example-analysis-workflow"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load configuration</span></span>
<span><span class="va">data_K562</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_json_file.html">read_json_file</a></span><span class="op">(</span><span class="st">"pacbio_outputs.json"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Read structural variants</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">VariantAnnotation</span><span class="op">)</span></span>
<span><span class="va">sv_vcf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/readVcf-methods.html" class="external-link">readVcf</a></span><span class="op">(</span><span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.phased_sv_vcf</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 3. Filter for specific SV types</span></span>
<span><span class="va">sv_dels</span> <span class="op">&lt;-</span> <span class="va">sv_vcf</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html" class="external-link">info</a></span><span class="op">(</span><span class="va">sv_vcf</span><span class="op">)</span><span class="op">$</span><span class="va">SVTYPE</span> <span class="op">==</span> <span class="st">"DEL"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># 4. Access methylation data</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span>
<span><span class="va">methylation</span> <span class="op">&lt;-</span> <span class="fu">import</span><span class="op">(</span><span class="va">data_K562</span><span class="op">$</span><span class="va">humanwgs_singleton.cpg_combined_bed</span><span class="op">)</span></span></code></pre></div>
<p>This structured approach allows you to efficiently access and analyze
the comprehensive genomic data produced by the PacBio WDL pipeline while
handling the path translation issues seamlessly.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Scott Furlan.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>


<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({


    apiKey: 'YOUR_API_KEY',
    indexName: 'pacbiowdlR',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
